Compile/src/Producers/LeptonSFProducer.cc

```diff
@@ -440,24 +441,25 @@ void LeptonTriggerSFProducer::Produce(ZJetEvent const& event,
                                  ZJetProduct& product,
                                  ZJetSettings const& settings) const
 {
-    if(product.m_zValid){
-        product.m_weights["zl1TriggerSFWeight"] = 1/GetScaleFactor(0, *product.m_zLeptons.first);
-        product.m_weights["zl2TriggerSFWeight"] = 1/GetScaleFactor(0, *product.m_zLeptons.second);
-        product.m_weights["leptonTriggerSFWeight"] =
-            1/(1-(1-GetScaleFactor(0, *product.m_zLeptons.first)) * (1-GetScaleFactor(0, *product.m_zLeptons.second)));
+    if(product.m_zValid) {
+        float sf1 = 1 / GetScaleFactor(0, *product.m_zLeptons.first);
+        float sf2 = 1 / GetScaleFactor(0, *product.m_zLeptons.second);
+        product.m_weights["zl1TriggerSFWeight"] = sf1;
+        product.m_weights["zl2TriggerSFWeight"] = sf2;
+        product.m_weights["leptonTriggerSFWeight"] = sf1 * sf2;
         if (settings.GetLeptonSFVariation() == true) {
-            product.m_weights["leptonTriggerSFWeightUp"] =
-                //(1-(1-GetScaleFactor(2, -1, *product.m_zLeptons.first)) * (1-GetScaleFactor(2, -1, *product.m_zLeptons.second)))
-                1/(1-
-                  (1-GetScaleFactor(1, *product.m_zLeptons.first)  + 0.005*GetScaleFactor(0, *product.m_zLeptons.first))
-                * (1-GetScaleFactor(1, *product.m_zLeptons.second) + 0.005*GetScaleFactor(0, *product.m_zLeptons.second))
-                );
-            product.m_weights["leptonTriggerSFWeightDown"] =
-                //(1-(1-GetScaleFactor(2, 1, *product.m_zLeptons.first)) * (1-GetScaleFactor(2, 1, *product.m_zLeptons.second)))
-                1/(1-
-                  (1-GetScaleFactor(-1, *product.m_zLeptons.first)  - 0.005*GetScaleFactor(0, *product.m_zLeptons.first))
-                * (1-GetScaleFactor(-1, *product.m_zLeptons.second) - 0.005*GetScaleFactor(0, *product.m_zLeptons.second))
-                );
+            // up has -1 since it's inside the denominator, down +1
+            float sf1u = 1 / GetScaleFactor(-1, *product.m_zLeptons.first);
+            float sf1d = 1 / GetScaleFactor(+1, *product.m_zLeptons.first);
+            float sf2u = 1 / GetScaleFactor(-1, *product.m_zLeptons.second);
+            float sf2d = 1 / GetScaleFactor(+1, *product.m_zLeptons.second);
+            product.m_weights["zl1TriggerSFWeightUp"] = sf1u;
+            product.m_weights["zl1TriggerSFWeightDown"] = sf1d;
+            product.m_weights["zl2TriggerSFWeightUp"] = sf2u;
+            product.m_weights["zl2TriggerSFWeightDown"] = sf2d;
+            // proper error propagation needed
+            product.m_weights["leptonTriggerSFWeightUp"] = sf1u * sf2u;
+            product.m_weights["leptonTriggerSFWeightDown"] = sf1d * sf2d;
```

Comment:
Same here. I don't know why these two scale factors are combined like this, but this needs to be cleared.